autox/environment/nncf/quantizer_tracing.py

```diff
@@ -0,0 +1,511 @@
+            return _qid
+
+def uptrace_quantize_nncfnode(nncfgraph, nncfnode, uptrace_node_list):
+    g=nncfgraph
+    _nodekey = g.get_node_key_by_id(nncfnode.node_id)
+    if "Quantizer" in _nodekey:
+        # add quantize node to the list
+        uptrace_node_list.append(nncfnode)
+    else:
+        # uptrace for non-quantize node
+        input_nncfnodes = g.get_previous_nodes(nncfnode)
+        for input_nncfnode in input_nncfnodes:
+            uptrace_quantize_nncfnode(g, input_nncfnode, uptrace_node_list)
+    return uptrace_node_list
+
+def get_non_weight_quantizer_id(nncfgraph, nncfnode):
+    g=nncfgraph
+    nodekey=g.get_node_key_by_id(nncfnode.node_id)
+    if 'quantize' not in nodekey:
+        raise ValueError("This is not a quantize node")
+    
+    # Assumption: non-weight quantizer id usually follows
+    # the scope of predecessor node (from observation thus far)
+    innodes = g.get_previous_nodes(nncfnode)
+
+    if len(innodes) != 1:
+        raise ValueError("Special Case, Assumption breaks, more than a single parent nodes")
+
+    innodekey=g.get_node_key_by_id(innodes[0].node_id)
+    in_nxnode=g.get_nx_node_by_key(innodekey)
+
+    return NonWeightQuantizerId(in_nxnode['op_exec_context'].input_agnostic)
+
+def NWQId_exist_Qctrl(NWQId, quantization_controller):
+    for key in quantization_controller.non_weight_quantizers.keys():
+        if NWQId.__str__() == key.__str__():
+            return True
+    return False
+# End of Helper Functions ==================================
+
+
+# Assumption: All GEMM has only two parent nodes (1) parameters loading node (2) Any Operation node in DAG
+
+# Extra Info:
+# NNCFNode is a base class that capture node_id and IA_OP_CTX. 
+# NNCFGraph is networkx digraph, each node is nx_node. To convert nx_node to NNCFNode, use NNCFGraph._nx_node_to_nncf_node(nx_node)
+# we refer node as nncfnode and nx_node as networkx node
+
+# gemm_module as key
+# WQ => (nodekey, quantizer)
+# NWQ => [(nodekey, quantizer),(nodekey, quantizer), ...]
+# skiplist store the nncf_module that only exist for training purpose; tracer is not able to prune the nncf_modules to contain eval-only module
+gemm_skiplist =[
+    'ICNet/CascadeFeatureFusion[cff42]/NNCFConv2d[classifier]',
+    'ICNet/CascadeFeatureFusion[cff421]/NNCFConv2d[classifier]'
+]
+qid_skiplist= [
+    'ICNet/CascadeFeatureFusion[cff42]/NNCFConv2d[classifier]module_weight',
+    'ICNet/CascadeFeatureFusion[cff421]/NNCFConv2d[classifier]module_weight',
+]
+def get_gemm_with_input_quantizers(quantization_controller, quantized_network):
+    g=quantized_network.get_graph() # compressed nncf graph instance
+
+    gemm_scope_list= quantized_network.get_nncf_module_scopes()
+
+    gemm_quantizer_dict=OrderedDict()
+
+    print("= Trace input quantizer to GEMM =======")
+
+    for gemm in gemm_scope_list:
+        
+        if str(gemm) in gemm_skiplist:
+            continue
+
+        print("GEMM Scope | {}".format(str(gemm)))
```

Comment:
All library code should be without prints. Please replace it with logger

Reply:
agree :)

Reply:
Low priority

Reply:
quantizer_tracing.py has been removed. Functionality is achieved via NNCF utility